ci/autoscaler/tasks/update-sdk/update_java_package.sh

```diff
@@ -1,18 +1,63 @@
 #! /usr/bin/env bash
+
+# This script is used to update the java version used in the bosh packaging
+# It downloads the given SAP Machine java distribution and updates the bosh packaging with the new version.
+# usage: ./update_java_package 21.0.3
+
 [ -n "${DEBUG}" ] && set -x
 set -euo pipefail
 
 script_dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
-# shellcheck source=vendor_package.sh
 source "${script_dir}/vars.source.sh"
-source "${script_dir}/vendor_package.sh"
 
-java_dir=${JAVA_DIR:-"${autoscaler_dir}/../java-release"}
-java_dir=$(realpath --canonicalize-existing "${java_dir}")
-export java_dir
+JAVA_VERSION=${1:-"21.0.3"} # default java version
+desired_major_version=${JAVA_VERSION%%.*}
+
+# identify current version
+current_java_version=$(find ./packages -type d -name "openjdk-*" -exec bash -c '
+                          directory_name="$1"
+                          version_number=${directory_name#*-}
+                          echo "${version_number}"
+                          ' bash {} \;)
+echo "Current Java Major version is ${current_java_version}"
+
+# Step 1 --> Download java...
+source "${script_dir}/download_java.sh" "${JAVA_VERSION}"
+
+# Step 2 --> update bosh java package references
+echo "updating bosh java packages..."
+# shellcheck disable=SC2038
+find . -type f ! -name "*.yml" ! -name "update_java_package.sh" ! -path '*/\.*' -exec grep -l "openjdk-${current_java_version}" {} \;| xargs sed -i "s/openjdk-${current_java_version}/openjdk-${desired_major_version}/g"
+
+mv ./packages/openjdk-"${current_java_version}" ./packages/openjdk-"${desired_major_version}"
+
+echo "  -- creating spec file"
+binary_name="sapmachine-jdk-${JAVA_VERSION}_linux-x64_bin.tar.gz"
```

Comment:
We need to also put the file into the blobstore, see https://bosh.io/docs/release-blobs/#adding-blob

Then the download will be taken care of by BOSH when building the release.